Chat_App/Core/Custom UI/ViewModel/ChatViewModel.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  ChatViewModel.swift
+//  test
+//
+//  Created by Ana Dzamelashvili on 5/23/23.
+//
+
+import Foundation
+import UIKit
+
+protocol ChatViewModelDelegate: AnyObject {
+    func messagesLoaded()
+}
+
+class ChatViewModel {
+    weak var delegate: ChatViewModelDelegate?
+    private let coreDataManager = CoreDataManager()
+    private lazy var sentMessages = [MessageEntity]()
+  
+    
+    func filteredMessages(currentUser: Int) -> [MessageEntity] {
+        var filtered = [MessageEntity]()
+        
+        sentMessages.forEach { message in
+            if message.userId == currentUser || !message.failedToSend {
+                filtered.append(message)
+            }
+        }
+        return filtered
+    }
+
+    
+    //MARK: - Load messages from coreData
+    func loadMessages() {
+        if let messages = coreDataManager.loadObjects(with: coreDataManager.request) {
```

Comment:
see what I was talking about in CoreDataManager? 
If `loadObjects(with:)` would be a method that throws, you could be doing there just 
do {
sendMessages = try coreDataManager.loadObjects(with:)
delegate?.messagesLoaded()
} catch error as? CoreDataManager.InternalError { // syntax is pseudo, but you can check documentation on that
delegate.sendError(error)
}

Reply:
and no optionals unwrapping while working with your manager. You'd be able to throw errors at your delegate that would be showing certain views in that case or maybe an UIAlertController